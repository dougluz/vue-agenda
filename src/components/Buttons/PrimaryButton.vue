<template>
  <button class="bg-light-red h-8 pt-2 pb-2 pl-4 pr-4 rounded-2xl">
    <slot> Salvar </slot>
  </button>
</template>

<style scoped>
.primary-button {
  @apply bg-light-red text-white text-sm font-medium h-8 pt-2 pb-2 pl-4 pr-4 rounded-2xl;
}
</style>
