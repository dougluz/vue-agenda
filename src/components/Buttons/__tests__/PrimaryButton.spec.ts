import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import PrimaryButton from "../PrimaryButton.vue";

describe("PrimaryButton", () => {
  describe("when rendering the default version", () => {
    it("should match the snapshot", () => {
      const wrapper = mount(PrimaryButton);

      expect(wrapper.html()).toMatchSnapshot();
    });

    it("should show the default text", () => {
      const wrapper = mount(PrimaryButton);

      expect(wrapper.text()).toBe("Salvar");
    });
  });
  describe("when passing a custom text in the slot", () => {
    it("should show the custom text", () => {
      const wrapper = mount(PrimaryButton, {
        slots: {
          default: "Custom text",
        },
      });

      expect(wrapper.text()).toBe("Custom text");
    });
  });
});
