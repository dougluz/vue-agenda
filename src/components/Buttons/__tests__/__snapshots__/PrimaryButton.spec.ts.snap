// Vitest Snapshot v1

exports[`PrimaryButton > when rendering the default version > should match the snapshot 1`] = `"<button class=\\"bg-light-red h-8 pt-2 pb-2 pl-4 pr-4 rounded-2xl\\" data-v-2da1bdcc=\\"\\"> Salvar </button>"`;
